src/markdown.ts

```diff
@@ -74,6 +75,12 @@ export function createMarkdown(options: ResolvedOptions) {
     if (options.frontmatter) {
       const { head, frontmatter } = frontmatterPreprocess(data || {}, options)
       scriptLines.push(`const frontmatter = ${JSON.stringify(frontmatter)}`)
+      if (options.exposeFrontmatter) {
+        if (defineExposeRE.test(hoistScripts.scripts.join('')))
+          console.warn(`[vite-plugin-md] frontmatter couldn't be auto exposed, since defineExpose was already called in ${id}.`)
```

Comment:
We could silently disable the expose as they are explicitly set the defineExpose.